.github/workflows/rust.yml

```diff
@@ -33,6 +33,38 @@ jobs:
           components: rustfmt
       - run: cargo fmt --all -- --check
 
+  benchmark:
+    name: Performance regression check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@nightly
+      - name: Run benchmark
+        run: |
+          set -eo pipefail
+          cargo +nightly bench | tee output.txt
+      # Download previous benchmark result from cache (if exists)
+      - run: cat output.txt
+      - name: Download previous benchmark data
+        uses: actions/cache@v3
+        with:
+          path: ./cache
+          key: ${{ runner.os }}-benchmark
```

Comment:
This won't work. GitHub will not override the cache when it exists. This does not guarantee that the benchmark gets compared with the previous version.

Reply:
You are right. Wondering how to fix it. (It is the from the action recommend way to store your benchmark data)

Reply:
I have remove the benchmark action for now. So the bench results can only be viewed at the logfile.